script.js

```diff
@@ -1,4 +1,60 @@
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var lowercaseinput = input.toLowerCase(input);
+
+  var number = chooseobject();
+
+  // lets assign an object to each number generated
+  var systemobject = "";
+
+  if (number == 1) {
+    systemobject = "scissors";
+  }
+  if (number == 2) {
+    systemobject = "paper";
+  }
+  if (number == 3) {
+    systemobject = "stone";
+  }
+
+  // lets define the default output
+  var myOutputValue =
+    "Ops, you lost. " +
+    "<br>" +
+    "Your " +
+    lowercaseinput +
+    " doesn't beat my " +
+    systemobject +
+    "!";
+
+  // lets define how a user can win
+  if (
+    (lowercaseinput == "reversed scissors" && systemobject == "stone") ||
+    (lowercaseinput == "reversed paper" && systemobject == "scissors") ||
+    (lowercaseinput == "reversed stone" && systemobject == "paper")
+  ) {
+    myOutputValue =
+      "Dang, you won!" +
+      "<br>" +
+      "Your " +
+      lowercaseinput +
+      " beat my " +
+      systemobject +
+      "!";
+  }
+
+  // lets define a draw
+  if (lowercaseinput == systemobject) {
+    myOutputValue =
+      "Ooo, you read my mind. I chose " + systemobject + " too. It's a draw!";
+  }
+
   return myOutputValue;
 };
+
+// lets generate three random numbers
+var chooseobject = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomNumber = Math.ceil(randomDecimal);
```

Comment:
Interesting choice of `Math.ceil` here, although that's not what I would have recommended for this random number generation logic.

Consider that in [line 56](https://github.com/CheongYuting/basics-scissors-paper-stone/blob/6bcc5598d5a9ec5c4d515214615ed8a571f8414f/script.js#L56), `randomDecimal` will be between 0 and 2.999999...

I assume in [line 57](https://github.com/CheongYuting/basics-scissors-paper-stone/blob/6bcc5598d5a9ec5c4d515214615ed8a571f8414f/script.js#L57), you believed rounding up will give you whole numbers from 1 to 3. Unfortunately, you missed the _very rare_ case where `randomDecimal` is still a flat 0. Rounding up the integer / whole number 0 still gives you 0, while only rounding up 0.0000...001 decimal gives you 1.

This also means you may also come across the _rare_ case in [lines 7 to 17](https://github.com/CheongYuting/basics-scissors-paper-stone/blob/6bcc5598d5a9ec5c4d515214615ed8a571f8414f/script.js#L7-L17) where `systemobject` is still `""`, because it doesn't match 1, 2, or 3.